Constantinople release (#58)
Joystream/helpdesk: proposals/README.md

As the `Council` is meant to be be the executive arm of the Joystream governance system, we are adding a proposal system that will allow the users to control most of the day to day decision making on the platform.

## Implemented Proposals

The list below contains all the proposal types currently available.

#### Text/signal Proposal
Although no action will happen if such a proposal is voted through, it provides a way for user to request changes, propose improvements, complaint about something, and in general voice their opinion on a matter. This will open a discussion, and Council Member can signal their approval or rejection through a vote. This can be used to notify the platform developers about key feature missing, highlight a topic of controversy, etc.

#### Funding Requests
In general, this proposal will include an amount, and a beneficiary. This can be used in to fund development, pay winners of competitions, bonus payments for a role, or anything else that requires minting new tokens to a specific individual or group.

#### Evict Storage Provider

> This allows users to make a proposal to fire a Storage Provider that does not perform the role satisfactory.

Reviewer: ```suggestion
This allows users to make a proposal to "fire" a Storage Provider that is not performing the role satisfactorily.
```

The parameters below are specific to each proposal type, and the values are chosen to balance the importance and risks associated with each of them.

- **Voting Period [blocks]** - the maximum amount of blocks where voting is open before a proposal expires
- **Grace Period [blocks]** - the amount of blocks after a proposal is approved until it as implemented on chain
- **Approval Quorum [%]** - the amount of votes required to be cast before a proposal _can_ be approved
- **Approval Threshold [%]** - the required threshold ratio of cast votes for `approve`, relative to those that vote `abstain`, `reject` or `slash`
- **Slashing Quorum [%]** -  the amount of votes required to be cast before a proposal _can_ lead to slashing the stake of the proposer
- **Slashing Threshold [%]** - the required threshold ratio of cast votes that slash relative to those that vote approve, abstain or reject.
- **Proposal Stake [tJOY]** - the required stake to create a proposal of this type

There are also some general parameters that applies equally to all proposals, such as fees, length of text allowed, the maximum number of proposals currently in the `active` stage, etc.

## Proposal Flow

> Any member of the platform with sufficient amount of tokens to stake, can create a proposal. If the proposal includes values that are within the allowable range, and there are fewer `active` proposals than what is allowed, the proposal was proceed to the `active` state where the council can vote on it.

Reviewer: ```suggestion
Any member of the platform with a sufficient quantity of tokens to stake can create a proposal. If the proposal includes values that are within the allowable range, and there are fewer `active` proposals than permitted, the proposal was proceed to the `active` state where the council can vote on it.
```
